### PR TITLE
Add Keyvault support for db credentials

### PR DESCRIPTION
**Pull request by bryevdv**
This PR cribs heavily from 

https://azure.microsoft.com/en-us/resources/samples/key-vault-python-manage/

Separately:

* `team-standup` KeyVault has been added in the Azure Portal with `database-url` secret
* `KEY_VAULT_URI` env var was added to application service settings 
* System assigned Managed Identity was enabled for the app service

It's unclear to me if there are other necessary steps I have missed, so comments are welcome.

### CODE REVIEW

File: app/config.py [hunk 1]
```diff
@@ -1,5 +1,25 @@
...
+    if "APPSETTING_WEBSITE_SITE_NAME" in os.environ:
+        credentials = MSIAuthentication(resource='https://vault.azure.net')
+        key_vault_client = KeyVaultClient(credentials)
+
+        key_vault_uri = os.environ.get("KEY_VAULT_URI")
+
+        return key_vault_client.get_secret(
+            key_vault_uri,
+            "database-url",
+            ""
```
Comment (lmazuel) [inline app/config.py]: I believe this empty string deserves a "this is latest"

File: requirements.txt [hunk 2]
```diff
@@ -2,4 +2,5 @@ flask_migrate==2.3.0
 flask_script==2.0.6
 psycopg2==2.7.5
 flask_bcrypt==0.7.1
-applicationinsights==0.11.8
\ No newline at end of file
+applicationinsights==0.11.8
+azure-mgmt-keyvault==1.1.0
```
Comment (lmazuel) [inline requirements.txt]: You actually use azure-keyvault here, not azure-mgmt-keyvault

File: app/config.py [hunk 3]
```diff
@@ -1,5 +1,25 @@
 import os
 
+from azure.keyvault import KeyVaultClient
+from msrestazure.azure_active_directory import MSIAuthentication
+
+def _get_database_uri():
+
+    if "APPSETTING_WEBSITE_SITE_NAME" in os.environ:
```
Comment (lmazuel) [inline app/config.py]: To make this clearer, I try to externalize this test in a function:
```python
def _is_app_service():
    return 'APPSETTING_WEBSITE_SITE_NAME' in os.environ
```
I think it makes it simpler to read

File: app/config.py [hunk 4]
```diff
@@ -1,5 +1,25 @@
...
+        credentials = MSIAuthentication(resource='https://vault.azure.net')
+        key_vault_client = KeyVaultClient(credentials)
+
+        key_vault_uri = os.environ.get("KEY_VAULT_URI")
```
Comment (lmazuel) [inline app/config.py]: If you get `None` terrible things happen :)

File: app/config.py [hunk 5]
```diff
@@ -1,5 +1,25 @@
 import os
 
+from azure.keyvault import KeyVaultClient
+from msrestazure.azure_active_directory import MSIAuthentication
+
+def _get_database_uri():
+
+    if "APPSETTING_WEBSITE_SITE_NAME" in os.environ:
+        credentials = MSIAuthentication(resource='https://vault.azure.net')
```
[the remark this answers is not in the document]
Comment (bryevdv) [reply app/config.py]: addressed in https://github.com/bryevdv/Flask-React-Postgres/pull/2/commits/a467dc74fd6d1c3dd44f60e5aeaa8e6208eb7fdc

File: app/config.py (hunk 1 above)
Comment (bryevdv) [reply app/config.py]: addressed in https://github.com/bryevdv/Flask-React-Postgres/pull/2/commits/a467dc74fd6d1c3dd44f60e5aeaa8e6208eb7fdc

File: app/config.py (hunk 3 above)
Comment (bryevdv) [reply app/config.py]: addressed in https://github.com/bryevdv/Flask-React-Postgres/pull/2/commits/a467dc74fd6d1c3dd44f60e5aeaa8e6208eb7fdc

File: requirements.txt (hunk 2 above)
Comment (bryevdv) [reply requirements.txt]: fixed in https://github.com/bryevdv/Flask-React-Postgres/pull/2/commits/a467dc74fd6d1c3dd44f60e5aeaa8e6208eb7fdc

File: app/config.py (hunk 4 above)
Comment (lmazuel) [reply app/config.py]: Right now, you would get a `ValidationError: vault_base_url can not be None` from `get_secret`

Usually, I see something like this:
```python
key_vault_uri = os.environ.get("KEY_VAULT_URI", "PUT YOUR KV BASE URL HERE")
```

which ends in the worst scenario with something like  `"PUT YOUR KV BASE URL HERE" is not a valid url` :)

I'm not sure what is the best honestly, as long as you feel as a developer that you can get what's wrong fast enough (i.e. "I clone and ran this sample without giving in a thinking, what's wrong?")

Comment (bryevdv) [reply app/config.py]: This seems reasonable, I was mostly wondering if there was any standard mechanism that would e.g. surface orderly failures in an obvious way in the Portal when it was used. It seems that there is not, so I will update with this.

Comment (bryevdv) [reply app/config.py]: addressed in https://github.com/bryevdv/Flask-React-Postgres/pull/2/commits/c64e1d8aa930899d149bf7fdfab583ece65dcf93

**Comment by bryevdv** [comment]
OK, will merge this and see how it goes

**Pull request merged**